components/scream/src/control/tests/sc_tests.cpp

```diff
@@ -397,6 +407,15 @@ TEST_CASE ("recreate_mct_coupling")
   auto p_int_h            = p_int_f.get_view<Real**,Host>();
   auto phis_h             = phis_f.get_view<Real*,Host>();
 
+  // We need a copy of precip to check export is correct, since precip will be zeroed out after do_export is called.
+  // TODO: This step can be undone once we have the AD take care of resetting precip fluxes.
+  auto precip_liq_surf_copy_f = precip_liq_surf_f.clone(); 
+  auto precip_ice_surf_copy_f = precip_ice_surf_f.clone(); 
+  auto precip_liq_surf_copy_d = precip_liq_surf_copy_f.get_view<Real*>();
+  auto precip_ice_surf_copy_d = precip_ice_surf_copy_f.get_view<Real*>();
+  auto precip_liq_surf_copy_h = precip_liq_surf_copy_f.get_view<Real*,Host>();
```

Comment:
I'm not sure you need _copy_f and _copy_d since these are just used for storing data you're never going to change, then are only using for tests on host. Am I missing something? Note as well that other files use this same paradigm and should also be changed (in the unlikely event that my criticism is well placed).

Reply:
@tcclevenger can correct me if I'm wrong, but I'm pretty sure the REQUIRE statements are done on HOST but the actual export is done on DEVICE so we need both views, and to make sure they are synced.

Reply:
Ah, ok. I see now that the "_copy" version is needed to check that the data exported is the expected value of precip *before* sfc cpl zeroed things out. And I assume you get_view is assigning the pointer to the data rather than copying the data itself so we can't skip the _copy_f lines by directly assigning surf_f to copy_d and copy_h, right?

Reply:
Yes, we should be able to do it that way.  I think in order to do it I will have to first define 1d views on host and device in the test so that I can create and copy them.  I figured it was the same amount of code so I went with the simpler option since this is a test that will probably get replaced when @tcclevenger changes surface coupling to it's own process.  I'm happy either way.